idgen/idgen.go

```diff
@@ -139,7 +145,13 @@ func (bg *BalancedIdentityGenerator) Depth() int {
 }
 
 func (bg *BalancedIdentityGenerator) genID() (crypto.PrivKey, TrieKey, error) {
-	privKey, _, err := crypto.GenerateKeyPairWithReader(crypto.Ed25519, 0, bg.reader)
+	hash := sha256.New
+	info := []byte("hydra keys")
+	seed := bg.seed
+	salt := atomic.AddUint32(&bg.idgenCount, 1)
```

Comment:
as the access to the BalancedIdentityGenerator is already limited by a mutex, the atomic is not needed.